actix-identity/src/identity_errors.rs

```diff
@@ -0,0 +1,91 @@
+use actix_session::{SessionGetError, SessionInsertError};
+use actix_web::{cookie::time::error::ComponentRange, http::StatusCode, ResponseError};
+
+/// This error can occur during login attempts.
+#[derive(Debug)]
+pub struct LoginError(SessionInsertError);
+
+impl std::fmt::Display for LoginError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl std::error::Error for LoginError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        Some(&self.0)
+    }
+}
+
+impl ResponseError for LoginError {
+    fn status_code(&self) -> StatusCode {
+        StatusCode::UNAUTHORIZED
+    }
+}
+
+impl From<SessionInsertError> for LoginError {
+    fn from(error: SessionInsertError) -> Self {
+        Self(error)
+    }
+}
+
+/// This error describes all of the potential failures which can happen
+/// while retrieving an identity.
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum GetIdentityError {
+    /// This is an error which shouldn't occur, and indicates some kind of bug.
+    LostIdentityError,
+    /// This occurs whenever no identity is found in a session.
+    MissingIdentityError,
```

Comment:
For forward-compatibility - let's make this variant non-unit (i.e. they wrap a struct) and that struct is a unit struct marked with #[non_exhaustive]. It will allow us to add fields in the future in a non-breaking fashion.

Reply:
Ok! I am constructing and coercing it in the `get_identity` implementation. Is that the right way to hand it outward? `non_exhaustive` seems like it only cares if something outside the crate tries to construct it, but I want to be sure it's ok to do it in a neighbor module.

Reply:
Yes, the way you did is perfect 👌🏻 
We just need to do the same for `LostIdentityError` and then we are done.

Reply:
Do you have time to drive this to the finish line @esmevane? Or should we merge it as is and then add the remaining small changes to `LostIdentityError`?

Reply:
@LukeMathWalker Ah, mea culpa! I think the holidays got me. Stand by for this :)